OpenRA.Mods.Common/Activities/Air/FlyAttack.cs

```diff
@@ -48,8 +48,7 @@ public FlyAttack(Actor self, Target target, bool forceAttack, Color? targetLineC
 
 			// The target may become hidden between the initial order request and the first tick (e.g. if queued)
 			// Moving to any position (even if quite stale) is still better than immediately giving up
-			if ((target.Type == TargetType.Actor && target.Actor.CanBeViewedByPlayer(self.Owner))
-				|| target.Type == TargetType.FrozenActor || target.Type == TargetType.Terrain)
+			if (target.Type != TargetType.Invalid)
```

Comment:
It would be safer if we could add something like
```csharp
// The target may have just become hidden - make sure actors with FrozenUnderFog don't fall through the cracks
bool targetIsHiddenActor;
if (target.Type == TargetType.Actor && !target.Actor.CanBeViewedByPlayer(self.Owner))
	target = target.Recalculate(self.Owner, out targetIsHiddenActor);
```

before this. Outright removing the check means that the layout and some of the checks in `Tick` become redundant.

Reply:
A cleaner fix will be to handle this instead in `Tick` using something like
```csharp
// The target may become hidden in the same tick the FlyAttack constructor is called,
// causing lastVisibleTarget to remain uninitialized.
// Fix the fallback values based on the frozen actor properties
else if (target.Type == TargetType.FrozenActor && !lastVisibleTarget.IsValidFor(self))
{
	lastVisibleTarget = Target.FromTargetPositions(target);
	lastVisibleMaximumRange = attackAircraft.GetMaximumRangeVersusTarget(target);
	lastVisibleOwner = target.FrozenActor.Owner;
	lastVisibleTargetTypes = target.FrozenActor.TargetTypes;
}
```

immediately after the `if (!targetIsHiddenActor && target.Type == TargetType.Actor)` branch.

Reply:
Those suggestions don't fully resolve the problem, because the same problem exists for units that are not frozen under fog; it just manifests in a different way: The order just gets unceremoniously dropped even if it was valid at the moment it was given.

Also, at the moment, we are actually hemorrhaging actor positions whenever the activity is called at any time before the tick when the unit becomes hidden. Just queue a couple of move orders across the map and quickly queue an attack order, then hold shift.

The only way to fix this cleanly is to let the trait tick check for visibility changes and set the lastVisible* from there.

Reply:
>  The order just gets unceremoniously dropped even if it was valid at the moment it was given.

Yes, this is the expected behaviour designed into the activities. Changing this is well outside of the scope of a playtest bugfix.